src/Package/Impl/ExportDialog/ExportImageDialog.xaml.cs

```diff
@@ -0,0 +1,160 @@
+                    _originalHeight = UserHeight;
+                }
+                OnPropertyChanged("MaintainAspectRatio");
+            }
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        public ExportImageParameters GetExportParameters() {
+            ExportImageParameters exportImageParams = new ExportImageParameters();
+
+            exportImageParams.PixelHeight = UserHeight;
+            exportImageParams.PixelWidth = UserHeight;
+            exportImageParams.ViewPlot = ViewPlotAfterSaving;
+            return exportImageParams;
+        }
+
+        protected void OnPropertyChanged(string name) {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(name));
+        }
+
+        private void SaveButton_Click(object sender, RoutedEventArgs e) {
+            DialogResult = true;
+            Close();
+        }
+
+        private void CancelButton_Click(object sender, RoutedEventArgs e) {
+            DialogResult = false;
+            Close();
+        }
+
+        private void HeightTextbox_LostFocus(object sender, RoutedEventArgs e) {
+            TextBox textBox = (TextBox)sender;
+            int res = ValidateValues(textBox.Text);
+            if (res != -1) {
+                textBox.Text = res.ToString();
+                _prevValidHeight = UserHeight;
+            } else {
+                UserHeight = _prevValidHeight;
+            }
+
+            if (_maintainAspectRatio) {
+                int w = GetSizeForWidth();
+                UserWidth = w;
+                if (w == MIN || w == MAX) {
+                    int h = GetSizeForHeight();
+                    UserHeight = h;
+                }
+            }
+        }
+
+        private void WidthTextbox_LostFocus(object sender, RoutedEventArgs e) {
+            TextBox textBox = (TextBox)sender;
+            int res = ValidateValues(textBox.Text);
+            if (res != -1) {
+                textBox.Text = res.ToString();
+                _prevValidWidth = UserWidth;
+            } else {
+                UserWidth = _prevValidWidth;
+            }
+
+            if (_maintainAspectRatio) {
+                int h = GetSizeForHeight();
+                UserHeight = h;
+                if (h == MIN || h == MAX) {
+                    int w = GetSizeForWidth();
+                    UserWidth = w;
+                }
+            }
+        }
+
+        private int GetSizeForWidth() {
+            int width = UserHeight * _originalWidth / _originalHeight;
+            int w = ValidateValues(width.ToString());
+            return width = w;
+        }
+
+        private int GetSizeForHeight() {
+            int height = UserWidth * _originalHeight / _originalWidth;
+            int h = ValidateValues(height.ToString());
+            return height = h;
+        }
+
+        private int ValidateValues(string result) {
+            long l = 0;
+            bool isValid = long.TryParse(result, out l);
```

Comment:
`ValidateValues` is called over the string that was created from int and parses that string at the very beginning. What's the point?